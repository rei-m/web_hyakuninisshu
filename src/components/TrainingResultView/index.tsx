import * as React from 'react';
import { navigate } from 'gatsby';
import styled from '@src/styles/styled-components';
import { appTheme } from '@src/styles/theme';
import QuestionResultsSummary from '@src/components/QuestionResultsSummary';
import AppButton from '@src/components/AppButton';
import TweetButton from '@src/components/TweetButton';

export interface Props {
  totalCount: number;
  correctCount: number;
  averageAnswerSecond: number;
  onClickRestart: () => void;
}

const Container = styled.div`
  max-width: 380px;
  padding: ${({ theme }) => theme.spacing2x};
  margin: auto;
`;

const ShareButtonsWrapper = styled.div`
  text-align: center;
`;

const onClickBack = () => {
  navigate('/training', { replace: true });
};

const QuestionsResult: React.FC<Props> = ({ averageAnswerSecond, correctCount, totalCount, onClickRestart }) => (
  <Container>
    <QuestionResultsSummary
      title="正解数"
      value={`${correctCount} / ${totalCount}`}
      style={{ marginBottom: appTheme.spacing2x }}
    />
    <QuestionResultsSummary
      title="平均回答時間"
      value={`${averageAnswerSecond}秒`}
      style={{ marginBottom: appTheme.spacing4x }}
    />
    <ShareButtonsWrapper>
      <TweetButton text={`百人一首で ${totalCount}問中 ${correctCount}問 正解しました！`} />
    </ShareButtonsWrapper>
    {correctCount !== totalCount && (
      <AppButton
        label="間違えた歌の練習をする"
        icon="refresh"
        type="normal"
        onClick={onClickRestart}
        data-test="restart"
        style={{ margin: appTheme.spacing2x }}
      />
    )}
    <AppButton label="メニューに戻る" icon="arrow_back" type="normal" onClick={onClickBack} data-test="back" />
  </Container>
);

export default QuestionsResult;
