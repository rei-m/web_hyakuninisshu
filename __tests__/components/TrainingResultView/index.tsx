import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import { shallow } from 'enzyme';
import TrainingResultView, { Props } from '@src/components/TrainingResultView';
import { sel } from '@test/helpers';

describe('<TrainingResultView />', () => {
  let baseProps: Props;

  beforeEach(() => {
    baseProps = {
      averageAnswerSecond: 2.5,
      correctCount: 8,
      onClickRestart: jest.fn(),
      totalCount: 10,
    };
  });

  it('should render component when has wrong', () => {
    const renderer = ReactTestRenderer.create(<TrainingResultView {...baseProps} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should render component when has not wrong', () => {
    const props = { ...baseProps, correctCount: 10 };
    const renderer = ReactTestRenderer.create(<TrainingResultView {...props} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should fire onClickRestart when restart clicked', () => {
    const wrapper = shallow(<TrainingResultView {...baseProps} />).dive();
    wrapper.find(sel('restart')).simulate('click');
    expect(baseProps.onClickRestart).toHaveBeenCalled();
  });
});
