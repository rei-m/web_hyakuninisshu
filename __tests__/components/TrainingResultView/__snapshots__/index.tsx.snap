// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrainingResultView /> should render component when has not wrong 1`] = `
<div
  className="sc-bxivhb iNMnvU"
>
  <div
    className="sc-bdVaJa jXieMO"
    style={
      Object {
        "marginBottom": "16px",
      }
    }
  >
    <div
      className="sc-bwzfXH ijMQBI"
    >
      正解数
    </div>
    <div
      className="sc-htpNat hHyKqi"
    >
      10 / 10
    </div>
  </div>
  <div
    className="sc-bdVaJa jXieMO"
    style={
      Object {
        "marginBottom": "32px",
      }
    }
  >
    <div
      className="sc-bwzfXH ijMQBI"
    >
      平均回答時間
    </div>
    <div
      className="sc-htpNat hHyKqi"
    >
      2.5秒
    </div>
  </div>
  <div
    className="sc-ifAKCX fVKAHz"
  >
    <a
      className="twitter-share-button"
      data-show-count="false"
      data-size="large"
      data-text="百人一首で 10問中 10問 正解しました！"
      data-url="https://hyakuninanki.net"
      href="https://twitter.com/share"
    >
      Tweet
    </a>
  </div>
  <button
    className="MuiButtonBase-root-32 MuiButton-root-6 MuiButton-outlined-14 Component-button-1 Component-normalButton-2"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-7"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-35 MuiSvgIcon-fontSizeSmall-42 Component-normalIcon-4"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
      メニューに戻る
    </span>
    <span
      className="MuiTouchRipple-root-44"
    />
  </button>
</div>
`;

exports[`<TrainingResultView /> should render component when has wrong 1`] = `
<div
  className="sc-bxivhb iNMnvU"
>
  <div
    className="sc-bdVaJa jXieMO"
    style={
      Object {
        "marginBottom": "16px",
      }
    }
  >
    <div
      className="sc-bwzfXH ijMQBI"
    >
      正解数
    </div>
    <div
      className="sc-htpNat hHyKqi"
    >
      8 / 10
    </div>
  </div>
  <div
    className="sc-bdVaJa jXieMO"
    style={
      Object {
        "marginBottom": "32px",
      }
    }
  >
    <div
      className="sc-bwzfXH ijMQBI"
    >
      平均回答時間
    </div>
    <div
      className="sc-htpNat hHyKqi"
    >
      2.5秒
    </div>
  </div>
  <div
    className="sc-ifAKCX fVKAHz"
  >
    <a
      className="twitter-share-button"
      data-show-count="false"
      data-size="large"
      data-text="百人一首で 10問中 8問 正解しました！"
      data-url="https://hyakuninanki.net"
      href="https://twitter.com/share"
    >
      Tweet
    </a>
  </div>
  <button
    className="MuiButtonBase-root-32 MuiButton-root-6 MuiButton-outlined-14 Component-button-1 Component-normalButton-2"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "margin": "16px",
      }
    }
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-7"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-35 MuiSvgIcon-fontSizeSmall-42 Component-normalIcon-4"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      間違えた歌の練習をする
    </span>
    <span
      className="MuiTouchRipple-root-44"
    />
  </button>
  <button
    className="MuiButtonBase-root-32 MuiButton-root-6 MuiButton-outlined-14 Component-button-1 Component-normalButton-2"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-7"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-35 MuiSvgIcon-fontSizeSmall-42 Component-normalIcon-4"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
      メニューに戻る
    </span>
    <span
      className="MuiTouchRipple-root-44"
    />
  </button>
</div>
`;
